test/Microsoft.Azure.WebJobs.Host.EndToEndTests/DispatchQueueEndToEndTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests
+{
+    public class DispatchQueueEndToEndTests : IDisposable
+    {
+        private static ConcurrentDictionary<string, int> _concurrent = new ConcurrentDictionary<string, int>();
+        private JobHostConfiguration _hostConfiguration;
+        private const string HOSTID = "testhost";
+        private CloudQueue _sharedQueue;
+        private CloudQueue _poisonQueue;
+
+        public DispatchQueueEndToEndTests()
+        {
+            _hostConfiguration = new JobHostConfiguration();
+            _hostConfiguration.AddExtension(new DispatchQueueTestConfig());
+            _hostConfiguration.HostId = HOSTID;
+            CloudStorageAccount sdkAccount = CloudStorageAccount.Parse(_hostConfiguration.StorageConnectionString);
+            CloudQueueClient client = sdkAccount.CreateCloudQueueClient();
+            _sharedQueue = client.GetQueueReference("azure-webjobs-shared-" + HOSTID);
+            _poisonQueue = client.GetQueueReference("azure-webjobs-poison-" + HOSTID);
+        }
+
+        [Fact]
+        // same trigger type, multiple functions
+        public async Task DispatchQueueBatchTriggerTest()
+        {
+            _hostConfiguration.TypeLocator = new FakeTypeLocator(typeof(SampleTrigger));
+
+            using (var host = new JobHost(_hostConfiguration))
+            {
+                _concurrent.AddOrUpdate("DispatchQueueTrigger", 0, (k, v) => 0);
+                _concurrent.AddOrUpdate("DispatchQueueTrigger2", 0, (k, v) => 0);
+
+                host.Start();
+
+                await Task.Delay(15000);
```

Comment:
> await Task.Delay(15000); [](start = 15, length = 25)

Woah! We should not have a 15 second delay in the test.  What exactly is this waiting on?